src/mrpro/operators/models/_TransientInversionRecovery.py

```diff
@@ -0,0 +1,190 @@
+
+    def forward(self, m0: torch.Tensor, t1: torch.Tensor, alpha: torch.Tensor):
+        """Apply Transient Inversion Recovery signal model.
+
+        Parameters
+        ----------
+        m0
+            equilibrium signal / proton density
+        t1
+            longitudinal relaxation time T1
+        alpha
+            flip angle
+
+        Returns
+        -------
+            signal with dimensions ((... times), coils, z, y, x)
+        """
+        t1 = torch.where(t1 == 0, 1e-10, t1)
+        t1_star = 1 / (1 / t1 - torch.log(torch.cos(torch.deg2rad(alpha))) / self.tr)
+        m0_star = m0 * t1_star / t1
+
+        signal = torch.zeros((len(self.signal_time_points), *m0.shape), dtype=torch.complex64)
+
+        if len(self.index_before_first_inversion) > 0:
+            # Time relative to first acquisition
+            time_since_fist_acquisition = (
+                self.signal_time_points[self.index_before_first_inversion] - self.first_adc_time_point
+            )[(...,) + (None,) * (m0.ndim)]
+
+            signal[self.index_before_first_inversion, ...] = (
+                m0_star + (m0 - m0_star) * torch.exp(-time_since_fist_acquisition / t1_star)
+            ).to(dtype=torch.complex64)
+
+        # Signal just before first inversion pulse
+        if self.first_adc_time_point is not None and self.first_adc_time_point < self.inversion_time_points[0]:
+            time = self.inversion_time_points[0] - self.first_adc_time_point
+            m0_plus = m0_star + (m0 - m0_star) * torch.exp(-time / t1_star)
+        else:
+            m0_plus = m0
+
+        for inversion_index, index_of_time_points in enumerate(self.index_between_inversions):
+            # Calculate signal at the beginning of acquisition after the inversion pulse
+            m0_tau = -m0_plus * (1 - 2 * torch.exp(-self.time_inversion_adc / t1))
+
+            if len(index_of_time_points) > 0:
+                # Inversion times relative to current inversion pulse + time between inversion pulse and start of
+                # acquisition
+                inversion_time = (
+                    self.signal_time_points[index_of_time_points]
+                    - self.inversion_time_points[inversion_index]
+                    - self.time_inversion_adc
+                )[(...,) + (None,) * (m0.ndim)]
+
+                signal[index_of_time_points, ...] = (
+                    m0_star - (m0_tau + m0_star) * torch.exp(-inversion_time / t1_star)
+                ).to(dtype=torch.complex64)
+
+            # Signal at the beginning of the next inversion pulse
+            time = (
+                self.inversion_time_points[inversion_index + 1]
+                - self.inversion_time_points[inversion_index]
+                - self.time_inversion_adc
+            )
+            m0_plus = m0_star - (m0_tau + m0_star) * torch.exp(-time / t1_star)
+
+        return (rearrange(signal, 't ... c z y x -> (... t) c z y x'),)
```

Comment:
this will flatten all other dims, I think this is not what we wan here, right?
If my m0, t1, etc is split by slices, it should return the signal split by slcies

Reply:
Do we really want to introduce this bug just because no one has ifxex it yet for the other signal models?

Reply:
> The signal models should just keep the shape of the other dimensions unchanged...

I am not so sure. Eventually we want to map from parameter_map -> kdata and kdata is of shape (other, coils, k2, k1, k0) and therefore we need to somehow flatten the other dimension. Simply appending new dimensions at the beginning doesn't sound like a great idea to me.

Reply:
But matching the order will be an issue. 
So is it (other T) or (t other) and how to convert from one to the other. 
The signal models and optimizers work with tensors, so you would need to do this reordering anyways. 



Compromise:
Always flatten the time dimension into the last other dimension.

So for the other dimensions
-  (slice, 1) would result in (slice, t) < almost- how I am using it in my code, I would just need to add a singleton dimension

- (other) in (other*T) <- current behavior

So it is 't ...  other ->... (other t)' or similar.